agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -40,6 +65,7 @@ def apply
           warn "failed to sync #{service_pod.name} at #{service_pod.deploy_rev}: #{error}"
           sync_state_to_master(current_state, error)
         else
+          @container_state_changed = false
```

Comment:
> This also kinda breaks without #2138 to raise on container start errors:

It's not fatally broken, though, because the `kontena service deploy` still fails with a `GridServiceInstanceDeployer::StateError: Service instance is not running, but stopped`, and a further `kontena service deploy` will still bump the `deploy_rev` and have the worker attempt to re-start it.